docs/error_codes.md

```diff
@@ -1,42 +1,103 @@
 ## Error codes
 
 ### General init related
-- `GS000`: `Could not finish initialization`
-- `GS001`: `Threshold needs to be defined`
+- **GS000: Could not finish initialization**
+
+- **GS001: Threshold needs to be defined**   
+Threshold is `equal to zero`, some error could happen during the Safe setup.
 
 ### General gas/ execution related
-- `GS010`: `Not enough gas to execute Safe transaction`
-- `GS011`: `Could not pay gas costs with ether`
-- `GS012`: `Could not pay gas costs with token`
-- `GS013`: `Safe transaction failed when gasPrice and safeTxGas were 0`
+- **GS010: Not enough gas to execute Safe transaction**  
+SafeTxGas is too high or gas limit is too low to execute the transaction, keep in mind that we require some gas, at least `2500` is needed for events and at least `500` to perform code until execution.  
+
+- **GS011: Could not pay gas costs with ether**   
+There are not enough funds of `ether` to execute the transaction.   
+Make sure that you have enough funds of `ether` or choose other token with enough funds setting it in `gasToken`.  
+
+- **GS012: Could not pay gas costs with token**  
+There are not enough funds of the token choosen on `gasToken` to execute the transaction.      
+Make sure you have enough of the gasToken or set it to `address(0)` if you want use ether.  
+
+- **GS013: Safe transaction failed when gasPrice and safeTxGas were 0**  
+Transaction is not succesful, this could be for due to multiple reasons.
 
 ### General signature validation related
-- `GS020`: `Signatures data too short`
-- `GS021`: `Invalid contract signature location: inside static part`
-- `GS022`: `Invalid contract signature location: length not present`
-- `GS023`: `Invalid contract signature location: data not complete`
-- `GS024`: `Invalid contract signature provided`
-- `GS025`: `Hash has not been approved`
-- `GS026`: `Invalid owner provided`
+- **GS020: Signatures data too short**  
+There are less signatures than the owners threshold.  
+Provide as many as signatures as the owners threshold.  
```

Comment:
This is less about the signatures provide, more about the length of the bytes data. For this it probably makes sense to further explain how the signature data is created.

Reply:
Meybe instead add an example, add a doc link that contains examples:  
```
The length of required signatures is less than 65*threeshold.
Each signature has a constant length of 65 bytes `{32-bytes r}{32-bytes s}{1-byte v}`. If more data is necessary it can be appended to the end. 
More information about signatures:   
https://docs.gnosis-safe.io/contracts/signatures
```
What do you think?